lib/controllers/Motor.cpp

```diff
@@ -1,38 +1,99 @@
 #include "Motor.h"
 #include <cmath>
 
-Motor::Motor(PinName pwm, PinName dir, bool inverted, int freqInHz, float limit) :
+Motor::Motor(PinName pwm, PinName dir, bool inverted, int freqInHz, float limit, t_motorType motorType) :
         m_pwm(pwm), m_dir(dir), m_inverted(inverted), m_limit(limit) {
 
-    // Set initial condition of PWM
-    m_pwm.period(1.0 / freqInHz);
-    m_pwm = 0.0;
- 
-    // Initial condition of output enables
-    m_dir = 0;
+    // Motor
+    if(motorType == motor){
```

Comment:
IMO it would be much cleaner to make compatible extensions of the Motor class for each servo type, or making an overarching abstract base class for both motors and servos, instead of making all of these cases into one class. 

Ideally in OOP you want to aim to have classes only have one responsibility, because otherwise there is a lot of unnecessary interdependence. Classes should be related through inheritance.

Reply:
If done thoughtfully, you then wouldn't need to handle multiple cases in different places that use them.

Reply:
Actually I think it's not very reasonable to have m_servo and bunch of servo related functions in the Motor class, they should be moved to the ServoMotor class. This way we can prob eliminate these type checks without too much change

Reply:
Actually...what is even the point of having a ServoMotor class? what does it encapsulate? We don't need ActuactorController to control the Servo since it does not need to be controlled by PID. Can we just separate the concept between Motor and Servo, they are controlled differently and I don't really see a point of having them adhere to a common interface

Reply:
Yeah absolutely, there is a case though where it would be useful to have an encoder and servo linked... for example a continuous encoder being used instead of a motor.